src/SourceGenerators/Bit.SourceGenerators/AutoInject/AutoInjectSourceGenerator.cs

```diff
@@ -4,13 +4,22 @@
 using System.Linq;
 using System.Text;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Text;
 
 namespace Bit.SourceGenerators;
 
 [Generator]
 public class AutoInjectSourceGenerator : ISourceGenerator
 {
+    private static readonly DiagnosticDescriptor NonPartialClassError = new DiagnosticDescriptor(id: "BITGEN001",
+                                                                                              title: "The class use AutoInject must be partial",
```

Comment:
This message has nothing to say about classes which their base class has auto injected member
@msynk Could you please provide better error message?

Reply:
sure, here:

`{CLASS_NAME} is not partial. The AutoInject attribute needs to be used only in partial classes.`

remember to extract the correct class name (for example if needed the base class name should be used).